### ISSUE TITLE
Crash "Collection contains no element matching the predicate" in `isHasOperatorInvoke`

### ISSUE THREAD

**Issue by haluzpav** [Kiolk/Detekt-rules#3]
Hello,

the rule causes Detekt to crash.

Error:

```
> Task :app:detektDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:detektDebug'.
> Analyzing .../ReportException.kt led to an exception.
  Location: com.github.kiolk.rules.UseInvokeForOperator.isHasOperatorInvoke(UseInvokeForOperator.kt:100)
  The original exception message was: Collection contains no element matching the predicate.
  Running detekt '1.23.7' on Java '17.0.12+0' on OS 'Mac OS X'
  If the exception message does not help, please feel free to create an issue on our GitHub page.
```

The problematic `ReportException.kt` file:

```kotlin
package banana

import com.google.firebase.Firebase
import com.google.firebase.crashlytics.FirebaseCrashlytics
import com.google.firebase.crashlytics.KeyValueBuilder
import com.google.firebase.crashlytics.crashlytics
import com.google.firebase.crashlytics.setCustomKeys
import timber.log.Timber
import kotlin.coroutines.cancellation.CancellationException

@Suppress("NOTHING_TO_INLINE") // inlined so Timber shows the source in tag
inline fun reportException(
    throwable: Throwable,
    message: String? = null,
    noinline customKeys: (KeyValueBuilder.() -> Unit)? = null,
) {
    Timber.e(throwable, message)
    if (throwable is CancellationException) throw throwable
    Firebase.crashlytics.recordException(throwable) {
        customKeys?.invoke(this)
        if (message != null) key("message", message)
    }
}

fun FirebaseCrashlytics.recordException(
    throwable: Throwable,
    customKeys: (KeyValueBuilder.() -> Unit)? = null,
) {
    customKeys?.let(::setCustomKeys)
    recordException(throwable)
}
```

**Comment by Kiolk** [comment]
Thank 🙏🏻 you for bringing this issue to my attention and providing a detailed description.
I have identified a problem and reproduced this behavior. I am currently working on a solution for this issue. I will release a new version and notice you once the problem has been fixed.

**Comment by Kiolk** [comment]
I have published the new version 1.0.6 with a fix. It will be available in Maven in a few minutes. The problem was [here](https://github.com/Kiolk/Detekt-rules/pull/4/files#diff-4cd899ee81ba684edf2330a86e64ab5c234d49c90b1b956337e6c54d06841e15L90). I just preplaced the call `findFirstFunction` that caused an error with a safety call to `firstOrNull`. 

Additionally, I made some improvements. I added your piece of code to tests this case.

Again, thank you for your report.
